app/src/main/java/com/stevesoltys/backup/activity/backup/CreateBackupActivityController.java

```diff
@@ -77,9 +93,16 @@ void showEnterPasswordAlert(Set<String> selectedPackages, Uri contentUri, Activi
                 .setMessage("You'll need this to restore your backup, so write it down!")
                 .setView(passwordTextView)
 
-                .setPositiveButton("Set password", (dialog, button) ->
+                .setPositiveButton("Set password", (dialog, button) -> {
+                    if (passwordTextView.getText().length() == 0) {
+                        Toast.makeText(parent, "Please enter a password", Toast.LENGTH_SHORT).show();
+                        dialog.cancel();
+                        showEnterPasswordAlert(selectedPackages, contentUri, parent);
```

Comment:
I didn't find how an empty password would disable encryption. It would just crash, so for now I added this little part forcing the user to enter a password.

Reply:
It seems it was broken at some point recently (probably by me, honestly) but it used to only encrypt the data if the secret key was available: https://github.com/stevesoltys/backup/blob/master/app/src/main/java/com/stevesoltys/backup/transport/component/provider/ContentProviderBackupComponent.java#L213